server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestPutIndexTemplateAction.java

```diff
@@ -28,13 +28,21 @@
 
 public class RestPutIndexTemplateAction extends BaseRestHandler {
 
+    public static final String DEPRECATION_WARNING = "Legacy index templates are deprecated and will be removed completely in a " +
+        "future version. Please use composable templates instead.";
+    private static final RestApiVersion DEPRECATION_VERSION = RestApiVersion.V_8;
```

Comment:
Is this going to change to `V_7` for the backport? Should this be `V_7` right now so that the REST API compat. header still gets the deprecation (given that it's being deprecated in 7.x also)?

Reply:
Changing to `V_7` (on the 7.x backport) is almost certainly the correct thing to do. The semantics of that, then, would be that here on `master` the legacy index templates are "merely deprecated" and separately also that on `7.x` they are also "merely deprecated".